[GPU] CUDA 12.9 build error

<!--- WARNING: This is an issue tracker. Before opening a new issue make sure you read https://github.com/PointCloudLibrary/pcl/blob/master/CONTRIBUTING.md#using-the-issue-tracker. -->
**Describe the bug**

There are build errors with CUDA 12.9. The build is successful with CUDA 12.8. 

```
/pcl/gpu/features/src/centroid.cu(129): error: more than one instance of "transform_reduce" matches the argument list:
            function template "_Tp cuda::std::__4::transform_reduce(_InputIterator, _InputIterator, _Tp, _BinaryOp, _UnaryOp)" (declared at line 33 of /usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h)
            function template "OutputType thrust::THRUST_200802_SM_1200_NS::transform_reduce(InputIterator, InputIterator, UnaryFunction, OutputType, BinaryFunction)" (declared at line 55 of /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/transform_reduce.inl)
            argument types are: (thrust::THRUST_200802_SM_1200_NS::zip_iterator<cuda::std::__4::tuple<thrust::THRUST_200802_SM_1200_NS::device_ptr<float4>, thrust::THRUST_200802_SM_1200_NS::counting_iterator<int, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default>>>, thrust::THRUST_200802_SM_1200_NS::zip_iterator<cuda::std::__4::tuple<thrust::THRUST_200802_SM_1200_NS::device_ptr<float4>, thrust::THRUST_200802_SM_1200_NS::counting_iterator<int, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default>>>, pcl::device::TupleDistCvt, cuda::std::__4::tuple<float, int>, thrust::THRUST_200802_SM_1200_NS::maximum<cuda::std::__4::tuple<float, int>>)
          transform_reduce(
          ^
/usr/local/cuda/targets/x86_64-linux/include/thrust/detail/transform_reduce.inl(41): note #3322-D: number of parameters of function template "thrust::THRUST_200802_SM_1200_NS::transform_reduce(const thrust::THRUST_200802_SM_1200_NS::detail::execution_policy_base<DerivedPolicy> &, InputIterator, InputIterator, UnaryFunction, OutputType, BinaryFunction)" does not match the call
                   OutputType transform_reduce(
                              ^
/usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h(60): note #3322-D: number of parameters of function template "cuda::std::__4::transform_reduce(_InputIterator1, _InputIterator1, _InputIterator2, _Tp)" does not match the call
  transform_reduce(_InputIterator1 __first1, _InputIterator1 __last1, _InputIterator2 __first2, _Tp __init)
  ^
/usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h(43): note #3322-D: number of parameters of function template "cuda::std::__4::transform_reduce(_InputIterator1, _InputIterator1, _InputIterator2, _Tp, _BinaryOp1, _BinaryOp2)" does not match the call
                                           constexpr _Tp transform_reduce(
                                                         ^
          detected during instantiation of "float3 pcl::device::getMaxDistance(const pcl::gpu::DeviceArray<PointT> &, const float3 &) [with PointT=float4]" at line 169

/pcl/gpu/features/src/centroid.cu(156): error: more than one instance of "transform_reduce" matches the argument list:
            function template "_Tp cuda::std::__4::transform_reduce(_InputIterator, _InputIterator, _Tp, _BinaryOp, _UnaryOp)" (declared at line 33 of /usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h)
            function template "OutputType thrust::THRUST_200802_SM_1200_NS::transform_reduce(InputIterator, InputIterator, UnaryFunction, OutputType, BinaryFunction)" (declared at line 55 of /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/transform_reduce.inl)
            argument types are: (thrust::THRUST_200802_SM_1200_NS::zip_iterator<cuda::std::__4::tuple<thrust::THRUST_200802_SM_1200_NS::permutation_iterator<thrust::THRUST_200802_SM_1200_NS::device_ptr<float4>, thrust::THRUST_200802_SM_1200_NS::device_ptr<int>>, thrust::THRUST_200802_SM_1200_NS::counting_iterator<int, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default>>>, thrust::THRUST_200802_SM_1200_NS::zip_iterator<cuda::std::__4::tuple<thrust::THRUST_200802_SM_1200_NS::permutation_iterator<thrust::THRUST_200802_SM_1200_NS::device_ptr<float4>, thrust::THRUST_200802_SM_1200_NS::device_ptr<int>>, thrust::THRUST_200802_SM_1200_NS::counting_iterator<int, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default, thrust::THRUST_200802_SM_1200_NS::use_default>>>, pcl::device::TupleDistCvt, cuda::std::__4::tuple<float, int>, thrust::THRUST_200802_SM_1200_NS::maximum<cuda::std::__4::tuple<float, int>>)
      thrust::tuple<float, int> res = transform_reduce(
                                      ^
/usr/local/cuda/targets/x86_64-linux/include/thrust/detail/transform_reduce.inl(41): note #3322-D: number of parameters of function template "thrust::THRUST_200802_SM_1200_NS::transform_reduce(const thrust::THRUST_200802_SM_1200_NS::detail::execution_policy_base<DerivedPolicy> &, InputIterator, InputIterator, UnaryFunction, OutputType, BinaryFunction)" does not match the call
                   OutputType transform_reduce(
                              ^
/usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h(60): note #3322-D: number of parameters of function template "cuda::std::__4::transform_reduce(_InputIterator1, _InputIterator1, _InputIterator2, _Tp)" does not match the call
  transform_reduce(_InputIterator1 __first1, _InputIterator1 __last1, _InputIterator2 __first2, _Tp __init)
  ^
/usr/local/cuda/targets/x86_64-linux/include/cuda/std/__numeric/transform_reduce.h(43): note #3322-D: number of parameters of function template "cuda::std::__4::transform_reduce(_InputIterator1, _InputIterator1, _InputIterator2, _Tp, _BinaryOp1, _BinaryOp2)" does not match the call
                                           constexpr _Tp transform_reduce(
                                                         ^
          detected during instantiation of "float3 pcl::device::getMaxDistance(const pcl::gpu::DeviceArray<PointT> &, const pcl::device::Indices &, const float3 &) [with PointT=float4]" at line 170

2 errors detected in the compilation of "/pcl/gpu/features/src/centroid.cu".
make[2]: *** [gpu/features/CMakeFiles/pcl_gpu_features.dir/build.make:91: gpu/features/CMakeFiles/pcl_gpu_features.dir/src/centroid.cu.o] Error 2
make[1]: *** [CMakeFiles/Makefile2:1222: gpu/features/CMakeFiles/pcl_gpu_features.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**To Reproduce**
1. Use the following dockerfile:
```
FROM nvcr.io/nvidia/cuda:12.9.0-devel-ubuntu24.04

RUN apt-get update && \
    apt-get install -y git build-essential cmake libeigen3-dev libvtk9-dev libboost-all-dev

RUN git clone https://github.com/PointCloudLibrary/pcl.git --depth 1 && \
    mkdir pcl/build && \
    cd pcl/build/ && \
    cmake -DBUILD_GPU=ON -DCUDA_ARCH_BIN=120 ..

WORKDIR /pcl/build
```

2. Build the dockerfile: 
```
docker buildx build -t pcl_cuda_arch_120 .
```

3. Try building:
```
docker run -it --rm pcl_cuda_arch_120
make
```

**Your Environment (please complete the following information):**

 - OS: Ubuntu 24.04
 - PCL Version: HEAD
 - CUDA Version: 12.9

--- a/gpu/features/src/centroid.cu
+++ b/gpu/features/src/centroid.cu
@@ -89,7 +89,7 @@ void pcl::device::compute3DCentroid(const DeviceArray<PointT>& cloud, float3& ce
     thrust::device_ptr<PointT> src_beg((PointT*)cloud.ptr());
     thrust::device_ptr<PointT> src_end = src_beg + cloud.size();
 
-    centroid = transform_reduce(src_beg, src_beg, PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
+    centroid = thrust::transform_reduce(src_beg, src_beg, PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
     centroid *= 1.f/cloud.size();
 }
 
@@ -105,7 +105,7 @@ void pcl::device::compute3DCentroid(const DeviceArray<PointT>& cloud, const Indi
         thrust::device_ptr<int> map_end = map_beg + indices.size();
 
 
-        centroid = transform_reduce(make_permutation_iterator(src_beg, map_beg),
+        centroid = thrust::transform_reduce(make_permutation_iterator(src_beg, map_beg),
             make_permutation_iterator(src_beg, map_end),
             PointT2float3<PointT>(), make_float3(0.f, 0.f, 0.f), PlusFloat3());
 
@@ -126,7 +126,7 @@ float3 pcl::device::getMaxDistance(const DeviceArray<PointT>& cloud, const float
     thrust::maximum<thrust::tuple<float, int>> op;
 
     thrust::tuple<float, int> res =
-        transform_reduce(
+        thrust::transform_reduce(
         make_zip_iterator(make_tuple( src_beg, cf )),
         make_zip_iterator(make_tuple( src_beg, ce )),
         TupleDistCvt(pivot), init, op);
@@ -153,7 +153,7 @@ float3 pcl::device::getMaxDistance(const DeviceArray<PointT>& cloud, const Indic
     thrust::tuple<float, int> init(0.f, 0);
     thrust::maximum<thrust::tuple<float, int>> op;
 
-    thrust::tuple<float, int> res = transform_reduce(
+    thrust::tuple<float, int> res = thrust::transform_reduce(
         make_zip_iterator(make_tuple( make_permutation_iterator(src_beg, map_beg), cf )),
         make_zip_iterator(make_tuple( make_permutation_iterator(src_beg, map_end), ce )),
         TupleDistCvt(pivot), init, op);

